### ISSUE TITLE
[C#] Implement new Concept Exercise: switch-statement

### ISSUE THREAD

**Issue by mikedamay** [exercism/v3#1633]
This issue describes how to implement the `conditionals-switch` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how and when to use `switch` expressions and statements in C#.

## Learning objectives

- Know how to use `switch` expressions
- Know how to use `switch` statements (discussion only).

## Out of scope

- pattern matching types
- pattern matching tuples

## Concepts

- `conditionals-switch`

## Prerequisites

- `enums` (for understanding linked documentation)

Story dependent.

### Hints

[switch expression][switch-expression] documentation provides an introduction to `switch`.

### After

[switch expression][switch-expression] documentation provides an introduction to `switch`.
[switch statement][switch-statement] documentation provides an introduction to `switch` statements.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[switch-expression]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression#basic-example
[switch-statement]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/switch
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer

**Comment by mikedamay** [comment]
We should consider using this exercise to show `switch` statements and introduce `switch` expressions with `pattern-matching-tuples`.

**Comment by mikedamay** [comment]
I will give this a go.

**Comment by mikedamay** [comment]
@ErikSchierboom Please give this your consideration when you get a chance.  

Instructions are [here](https://github.com/mikedamay/v3/blob/csharp/switch-statements/languages/csharp/exercises/concept/switch-statements/.docs/instructions.md) and some example code is [here](https://github.com/mikedamay/v3/blob/csharp/switch-statements/languages/csharp/exercises/concept/switch-statements/.meta/Example.cs)

**Comment by ErikSchierboom** [comment]
@mikedamay Love the theme and it works nicely!

**Comment by mikedamay** [comment]
closed by #1777
